embedded_fab/main.dart

```diff
@@ -0,0 +1,416 @@
+          title: Text('Sent'),
+        ),
+        BottomNavigationBarItem(
+          icon: Icon(Icons.archive),
+          title: Text('Archived'),
+        ),
+      ],
+      backgroundColor: Colors.white,
+      selectedItemColor: blue500,
+      unselectedItemColor: grey600,
+      currentIndex: _selectedIndex,
+      elevation: 5,
+      showUnselectedLabels: true,
+      type: BottomNavigationBarType.fixed,
+      onTap: (value) {
+        setState(() {
+          _selectedIndex = value;
+        });
+      },
+    );
+  }
+
+  void _onFABTap() {
+    Navigator.push(
+        context,
+        MaterialPageRoute(
+          builder: (context) {
+            return Scaffold(
+              appBar: AppBar(
+                actions: <Widget>[
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 16.0),
+                    child: IconButton(
+                      icon: Icon(Icons.send),
+                      onPressed: () {
+                        Navigator.pop(context);
+                      },
+                    ),
+                  )
+                ],
+                backgroundColor: Colors.white,
+                brightness: Brightness.light,
+                elevation: 0,
+                iconTheme: IconThemeData(color: grey800),
+              ),
+              body: Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: Column(
+                  children: <Widget>[
+                    TextField(
+                      decoration: const InputDecoration(
+                        border: OutlineInputBorder(),
+                        filled: false,
+                        hintText: 'To',
+                      ),
+                    ),
+                    SizedBox(height: 8),
+                    TextField(
+                      decoration: const InputDecoration(
+                        border: OutlineInputBorder(),
+                        filled: false,
+                        hintText: 'Subject',
+                      ),
+                    ),
+                    SizedBox(height: 8),
+                    TextField(
+                      decoration: const InputDecoration(
+                        border: OutlineInputBorder(),
+                        filled: false,
+                        hintText: 'Compose email',
+                      ),
+                      maxLines: 10,
+                    ),
+                    SizedBox(height: 8),
+                  ],
+                ),
+              ),
+              extendBody: true,
+            );
+          },
+          fullscreenDialog: true,
+        ));
+  }
+
+
+  List<Widget> _drawerOptions() {
+    return [
+      Text('Options', style: Theme.of(context).textTheme.display1),
+      Divider(),
+      ListTile(title: Text('Shape')),
+      RadioListTile(
+        title: Text('Flat'),
+        value: ShapeType.Flat,
+        groupValue: _selectedShape,
+        onChanged: (value) {
```

Comment:
I think this can just be: `onChanged: _setSelectedTile`

Reply:
Done